ui/perfherder/alerts/StatusDropdown.jsx

```diff
@@ -0,0 +1,272 @@
+    const regex = {
+      revisionHref: /{+\srevisionHref\s}+/g,
+      alertHref: /{+\salertHref\s}+/g,
+      alertSummary: /{+\salertSummary\s}+/g,
+    };
+
+    for (const word of template.split(' ')) {
+      if (regex[word]) {
+        newTemplate = newTemplate.replace(regex[word], replacement[word]);
+      }
+    }
+    return newTemplate;
+  };
+
+  fileBug = async () => {
+    const { alertSummary, repos } = this.props;
+    // TODO it seems like it'd make more sense to fetch this once and customize/cache it for future use rather than
+    // fetching this template each time someone clicks on 'file bug' - regardless of test framework
+    const { data, failureStatus } = await getData(
+      getApiUrl(
+        `/performance/bug-template/?framework=${alertSummary.framework}`,
+      ),
+    );
+    if (!failureStatus) {
+      const result = data[0];
+      // repos is an instance of RepositoryModel, accessed on the $rootScope
+      const repo = repos.find(repo => repo.name === alertSummary.repository);
+
+      const templateArgs = {
+        revisionHref: repo.getPushLogHref(
+          alertSummary.resultSetMetadata.revision,
+        ),
+        alertHref: `${window.location.origin}/perf.html#/alerts?id=${
+          alertSummary.id
+        }`,
+        alertSummary: getTextualSummary(alertSummary),
+      };
+      const template = this.fillTemplate(result.text, templateArgs);
+
+      const pushDate = moment(
+        alertSummary.resultSetMetadata.push_timestamp * 1000,
+      ).format('ddd MMMM D YYYY');
+
+      const bugTitle = `${getTitle(alertSummary)} regression on push ${
+        alertSummary.resultSetMetadata.revision
+      } (${pushDate})`;
+
+      window.open(
+        `${bzBaseUrl}/enter_bug.cgi?${createQueryParams({
+          cc: result.cc_list,
+          comment: template,
+          component: result.default_component,
+          product: result.default_product,
+          keywords: result.keywords,
+          short_desc: bugTitle,
+          status_whiteboard: result.status_whiteboard,
+        })}`,
+      );
+    }
+  };
+
+  copySummary = () => {
+    const summary = getTextualSummary(this.props.alertSummary, true);
+    // can't access the clipboardData on event unless it's done from react's
+    // onCopy, onCut or onPaste props
+    navigator.clipboard.writeText(summary).then(() => {});
+  };
+
+  toggle = state => {
+    this.setState(prevState => ({
+      [state]: !prevState[state],
+    }));
+  };
+
+  updateAndClose = async (event, params, state) => {
+    event.preventDefault();
+    this.changeAlertSummary(params);
+    this.toggle(state);
+  };
+
+  changeAlertSummary = async params => {
+    const { alertSummary, updateAlertSummary } = this.props;
+    // TODO error handling
+    await update(
+      getApiUrl(`${endpoints.alertSummary}${alertSummary.id}/`),
+      params,
+    );
+    updateAlertSummary({ ...alertSummary, ...params });
+  };
+
+  isResolved = alertStatus =>
```

Comment:
This feels like the sort of thing that would better live in some js file devoted to business logic around alerts, so it can be tested seperately.

Reply:
Maybe. With the new react-testing library I might be able to test this functionality without extracting it into another file.